sdk/modelsrepository/Azure.Iot.ModelsRepository/src/DtmiConventions.cs

```diff
@@ -7,6 +7,10 @@
 
 namespace Azure.Iot.ModelsRepository
 {
+    /// <summary>
+    /// <c>DtmiConventions</c> implements the core aspects of the IoT model repo conventions
+    /// which includes calculating a generic path from a DTMI.
+    /// </summary>
     internal static class DtmiConventions
     {
         private static readonly Regex s_validDtmiRegex = new Regex(@"^dtmi:[A-Za-z](?:[A-Za-z0-9_]*[A-Za-z0-9])?(?::[A-Za-z](?:[A-Za-z0-9_]*[A-Za-z0-9])?)*;[1-9][0-9]{0,8}$");
```

Comment:
Independent of this PR, can we get a comment on this Regex that describes how it works. The universal truth about regex is that the author knows what it does, but it can appear undecipherable to others. :'-)

Reply:
Will do.